### DISCUSSION TITLE
Install Python via the VS code server's web panel

### DISCUSSION THREAD

**Discussion by PanGeo-Code** [coder/code-server#4366] [Q&A] [answered]
hello, I run Code server in an docker container on unraid.

Unraid Docker: Runs VS code Server.

And i dont realy understand how i need to install python 3 via the webpanel or console of the docker container.

so i was hoping someone could help here or explain?

Thanks

**Comment by jsjoeio** [comment, marked as answer]
Hmm I'm not familiar with unraid but you could be able to edit your Dockerfile and add a line to install Python, right? Maybe try [this](https://stackoverflow.com/a/50339110/3015595)?

**Reply by lg-nineoneninethreenine** [reply]
If you are on Debian or Ubuntu, try this:
```bash
sudo apt update
sudo apt install python3 python3-pip
python3 -v
```

This _has_ to work, since most of the Docker images are Debian or Ubuntu images.

**Reply by jsjoeio** [reply]
@im-coder-lg you're the best! Always coming in with the best answers :D

**Reply by lg-nineoneninethreenine** [reply]
I didn't do anything, I just learnt Debian commands and apt packages.

**Reply by nooobcoder** [reply]
@jsjoeio yes ofc, the Docker image of code-server has Debian under the hood, so it should be installed with the steps as told by @im-coder-lg 
![Screenshot](https://user-images.githubusercontent.com/50350828/139685063-36a7b754-c98f-40df-9d6d-c2c0713d2114.png)

Good job @im-coder-lg

**Reply by lg-nineoneninethreenine** [reply]
Thanks @nooobcoder 😁 I wonder adding potential dependencies like Python inside the Dockerfile since if you upgrade your deployment, Python, and other apps like noVNC will be installed by default.

**Reply by PanGeo-Code** [reply]
> If you are on Debian or Ubuntu, try this:
> 
> ```shell
> sudo apt update
> sudo apt install python3 python3-pip
> python3 -v
> ```
> 
> This _has_ to work, since most of the Docker images are Debian or Ubuntu images.

i added this to the docker immage and it worked

**Comment by PanGeo-Code** [comment]
Thanks, for all the info, 

i have gotr it to work thanks.
